src/libcore/intrinsics.rs

```diff
@@ -565,8 +565,17 @@ extern "rust-intrinsic" {
     pub fn atomic_umax_rel<T>(dst: *mut T, src: T) -> T;
     pub fn atomic_umax_acqrel<T>(dst: *mut T, src: T) -> T;
     pub fn atomic_umax_relaxed<T>(dst: *mut T, src: T) -> T;
+
+    #[cfg(not(stage0))]
+    pub fn prefetch<T>(data: *const T, rw: i32, locality: i32, cache: i32);
```

Comment:
No, we do not expect intrinsics to map directly to the underlying LLVM API at all. In fact, doing that makes little sense given our strong desire to support multiple backends (Cretonne, maybe?).

If this wasn’t prefetch, I would have suggested adding a regular wrapper with a sane API, but I think this intrinsic might want to avoid having wrapper functions (just like transmute).